### PR TITLE
`MultiHeadAttention._compute_attention_mask()` always returns a bool …

### PR DESCRIPTION
**Pull request by hertschuh**
…tensor.

Previously, if an non-bool `attention_mask` was passed and no other mask was passed, the original `attention_mask` was returned unchanged.

Now, it is always cast to bool.

### CODE REVIEW

**Review by divyashreepathihalli — APPROVED** [review]
Thanks Fabien!
